pkg/cloud/services/network/subnets.go

```diff
@@ -469,10 +489,14 @@ func (s *Service) createSubnet(sn *infrav1.SubnetSpec) (*infrav1.SubnetSpec, err
 	}
 
 	subnet := &infrav1.SubnetSpec{
-		ID:               *out.Subnet.SubnetId,
+		// Preserve the original identifier, this won't be a valid subnet ID, which
+		// is stored in the ResourceID field.
```

Comment:
```suggestion
		// Preserve the original identifier, this won't be a valid subnet ID in case
		// of a managed subnet. The AWS identifier `subnet-xxx` is stored in the ResourceID field.
```